storage/trace.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"os"
+
+	"cloud.google.com/go/storage/internal"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	otelcodes "go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	OpenTelemetryTracingExpVar = "GO_STORAGE_EXPERIMENTAL_OTEL_TRACING"
+	defaultTracerName          = "cloud.google.com/go/storage"
+	gcpClientRepo              = "googleapis/google-cloud-go"
+	gcpClientArtifact          = "storage"
+)
+
+// isOTelTracingDevEnabled checks the development flag until experimental feature is launched.
+func isOTelTracingDevEnabled() bool {
+	return os.Getenv(OpenTelemetryTracingExpVar) == "true"
+}
+
+func tracer() trace.Tracer {
+	return otel.Tracer(defaultTracerName, trace.WithInstrumentationVersion(internal.Version))
+}
+
+// startSpan accepts SpanStartOption and is used to replace internal/trace/StartSpan.
```

Comment:
This comment is fine for drafting, but we should change it to something else before merging

Reply:
Not seeing any changes here yet

Reply:
Updated PTAL, thanks!